packages/api/src/usePaginatedFetch.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import useFetch from './useFetch';
+import {
+    TSocketEndpointNames,
+    TSocketAcceptableProps,
+    TSocketRequestPayload,
+    TSocketRequestQueryOptions,
+} from '../types';
+
+const usePaginatedFetch = <T extends TSocketEndpointNames>(name: T, ...props: TSocketAcceptableProps<T, true>) => {
+    const limit = 10;
+    const [offset, setOffset] = useState(0);
+
+    const prop = props?.[0];
+    const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestPayload<T>) : undefined;
+    const options = prop && 'options' in prop ? (prop.options as TSocketRequestQueryOptions<T>) : undefined;
+
+    const nextPage = () => setOffset(prev => prev + limit);
+
+    const resetPages = () => setOffset(0);
+
+    // @ts-expect-error aaa
+    const fetcher = useFetch(name, {
+        payload: {
+            ...payload,
+            offset,
+            limit,
+        },
+        options: {
+            ...options,
+            keepPreviousData: true,
+        },
+    });
+
+    return {
+        ...fetcher,
+        nextPage,
+        resetPages,
+    };
+};
+
+export default usePaginatedFetch;
```

Comment:
```suggestion
const usePaginatedFetch = <T extends TSocketPaginateableEndpointNames>(
    name: T,
    ...props: TSocketAcceptableProps<T, true>
) => {
    const prop = props?.[0];
    const payload = prop && 'payload' in prop ? (prop.payload as TSocketRequestPayload<T>) : undefined;
    const options = prop && 'options' in prop ? (prop.options as TSocketRequestQueryOptions<T>) : undefined;

    // @ts-expect-error The `limit` parameter is always present in
    // the `payload` for the paginateable endpoints.
    const limit: number = payload?.payload?.limit || 10;
    // @ts-expect-error The `offset` parameter is always present in
    // the `payload` for the paginateable endpoints.
    const [offset, setOffset] = useState<number>(payload?.payload?.offset || 0);

    // @ts-expect-error It's safe to ignore the TS error here since the
    // exact type of the payload is not determined at this point.
    const { remove, ...rest } = useFetch(name, {
        payload: { ...payload, offset, limit },
        options: { ...options, keepPreviousData: true },
    });

    const loadMore = useCallback(() => setOffset(prev => prev + limit), [limit]);

    const reset = useCallback(() => {
        remove();
        setOffset(0);
    }, [remove]);

    return {
        ...rest,
        remove,
        loadMore,
        reset,
    };
};
```

Reply:
Done :bow: